File: SDKs/amazonaws.com/appflow/2020-08-23/java/lib/src/main/java/openapisdk/models/shared/IncrementalPullConfig.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * IncrementalPullConfig
 *  Specifies the configuration used when importing incremental records from the source. 
**/
public class IncrementalPullConfig {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("datetimeTypeFieldName")
    public String datetimeTypeFieldName;
    public IncrementalPullConfig withDatetimeTypeFieldName(String datetimeTypeFieldName) {
        this.datetimeTypeFieldName = datetimeTypeFieldName;
        return this;
    }
}

File: SDKs/amazonaws.com/appflow/2020-08-23/python/src/sdk/models/shared/incrementalpullconfig.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class IncrementalPullConfig:
    r"""IncrementalPullConfig
     Specifies the configuration used when importing incremental records from the source. 
    """
    
    datetime_type_field_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('datetimeTypeFieldName') }})
    
